File: TesteTecnico.Application/Domain/Entities/EquipmentStateHistory.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace TesteTecnico.Application.Domain.Entities;

public class EquipmentStateHistory
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public DateOnly Date { get; set; }

    [ForeignKey("EquipmentStateId")]
    public virtual EquipmentState EquipmentState { get; set; } = null!;
    public Guid EquipmentStateId { get; set; }

    [ForeignKey("EquipmentId")]
    public virtual Equipment Equipment { get; set; } = null!;
    public Guid EquipmentId { get; set; }
}

File: testebackendestagiov3/src/main/java/machado/antonio/testebackendestagiov3/entities/EquipmentStateHistory.java
package machado.antonio.testebackendestagiov3.entities;

import java.time.LocalDateTime;
import org.springframework.stereotype.Component;

import jakarta.persistence.Column;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Component
@Entity
@Table(name = "equipment_state_history", schema = "operation")
public class EquipmentStateHistory {

	@EmbeddedId
	private EquipmentStateHistoryId id;

	@ManyToOne
	@JoinColumn(name = "equipment_id", insertable = false, updatable = false)
	private Equipment equipment;

	@ManyToOne
	@JoinColumn(name = "equipment_state_id", insertable = false, updatable = false)
	private EquipmentState equipmentState;

	@Column(nullable = false)
	private LocalDateTime date;

	// Getters and Setters:

	public EquipmentStateHistoryId getId() {
		return id;
	}

	public void setId(EquipmentStateHistoryId id) {
		this.id = id;
	}

	public LocalDateTime getDate() {
		return date;
	}

	public void setDate(LocalDateTime date) {
		this.date = date;
	}

}
